drivers/ethernet/eth_e1000.c

```diff
@@ -317,7 +316,11 @@ static void e1000_iface_init(struct net_if *iface)
 	LOG_DBG("done");
 }
 
-static struct e1000_dev e1000_dev;
+DEVICE_PCIE_INST_DECLARE(0);
```

Comment:
We can also convert driver initialization to unreadable macrobatic code to support more then one instance.

Reply:
Sure. But I'll leave that outside of the scope of this PR. It's an enhancement that could have been done to the e1000 driver even without this PR.